src/citrine/resources/delete.py

```diff
@@ -1,54 +1,53 @@
+import json
 from typing import List, Union, Tuple, Optional
 from uuid import UUID
 
-from gemd.entity.link_by_uid import LinkByUID
 from gemd.entity.base_entity import BaseEntity
-from gemd.util import writable_sort_order
+from gemd.entity.link_by_uid import LinkByUID
 
 from citrine._session import Session
 from citrine.resources.api_error import ApiError
+from citrine.resources.job import _poll_for_job_completion
 
-DELETE_SERVICE_MAX = 50  # Edit here to change service limit
 
-
-def _gemd_batch_delete(
+def _async_gemd_batch_delete(
         id_list: List[Union[LinkByUID, UUID, str, BaseEntity]],
         project_id: UUID,
         session: Session,
-        dataset_id: Optional[UUID] = None
+        dataset_id: Optional[UUID],
+        timeout: float = 2 * 60,
+        polling_delay: float = 1.0
 ) -> List[Tuple[LinkByUID, ApiError]]:
     """
-    Shared implementation of GEMD Batch deletion.
-
-    You may provide GEMD objects that reference each other, and the objects
-    will be removed in the appropriate order.
-
-    A failure will be returned if the object cannot be deleted due to an external
-    reference.
-
-    If an optional dataset_id is provided, deletes are restricted to only occur on objects
-    contained by that specific dataset.
-
-    You must have Write access on the datasets associated with the GEMD objects provided.
+    Shared implementation of Async GEMD Batch deletion.
 
-    If you wish to delete more than 50 objects, queuing of deletes requires that
-    the types of objects be known, and thus you _must_ provide ids in the form
-    of BaseEntities.
-
-    Also note that Attribute Templates cannot be deleted at present.
+    See documentation for _gemd_batch_delete. The only difference is that this version polls for
+    an asynchronous result and can tolerate a very long runtime that the synchronous version
+    cannot. Because this version can tolerate a long runtime, this versions allows for the
+    removal of attribute templates.
 
     Parameters
     ----------
-    id_list: List[Union[LinkByUID, UUID, str, BaseEntity]]
-        A list of the IDs of data objects to be removed. They can be passed
-        as a LinkByUID tuple, a UUID, a string, or the object itself. A UUID
-        or string is assumed to be a Citrine ID, whereas a LinkByUID or
-        BaseEntity can also be used to provide an external ID.
+    id_list: List[Union[LinkByUID, UUID]]
+        A list of the IDs of data objects to be removed. They can be passed either
+        as a LinkByUID tuple, or as a UUID. The latter is assumed to be a Citrine
+        ID, whereas the former can also be used to provide an external ID.
```

Comment:
Why the change in documentation?  The sig matches the older version
```suggestion
    id_list: List[Union[LinkByUID, UUID, str, BaseEntity]]
        A list of the IDs of data objects to be removed. They can be passed
        as a LinkByUID tuple, a UUID, a string, or the object itself. A UUID
        or string is assumed to be a Citrine ID, whereas a LinkByUID or
        BaseEntity can also be used to provide an external ID.

Reply:
That wasn't intentional - good catch.